test/mysql-cdc/table-in-mysql-schema.td

```diff
@@ -0,0 +1,49 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+$ set-sql-timeout duration=1s
+$ set-max-tries max-tries=20
+
+$ postgres-execute connection=postgres://mz_system:materialize@${testdrive.materialize-internal-sql-addr}
+ALTER SYSTEM SET enable_mysql_source = true
+
+#
+# Test that tables in the mysql schema are not replicated
+#
+
+> CREATE SECRET mysqlpass AS '${arg.mysql-root-password}'
+> CREATE CONNECTION mysql_conn TO MYSQL (
+    HOST mysql,
+    USER root,
+    PASSWORD SECRET mysqlpass
+  )
+
+$ mysql-connect name=mysql url=mysql://root@mysql password=${arg.mysql-root-password}
+
+$ mysql-execute name=mysql
+USE mysql;
+
+# Insert data pre-snapshot
+CREATE TABLE t1 (f1 BOOLEAN);
+INSERT INTO t1 VALUES (true), (false);
+
+! CREATE SOURCE mz_source
+  FROM MYSQL CONNECTION mysql_conn
+  FOR ALL TABLES;
+contains:No tables found
+
+! CREATE SOURCE mz_source
+  FROM MYSQL CONNECTION mysql_conn
+  FOR TABLES (mysql.timezone);
+contains:No tables found
+
+! CREATE SOURCE mz_source
+  FROM MYSQL CONNECTION mysql_conn
+  FOR SCHEMAS (mysql);
+contains:referenced tables use unsupported types
```

Comment:
@rjobanp: Are we ok with this or should we expect `No tables found`?

Reply:
I think if it was explicitly asked for in FOR SCHEMAS and FOR TABLES we should allow them to ingest tables from any schema, including the system schemas we ignore with FOR ALL TABLES. But I could be convinced that the semantics of FOR SCHEMAS should be different than FOR TABLES -- wdyt @morsapaes ?

Reply:
Ok. I assume this is currently the case since we get `referenced tables use unsupported types`.